add external spi-nor flash will build fail

**Describe the bug**
I add spi-nor flash for nrf52832 board, want to the external spi flash & nrf52832 internal flash work same times. but build error:
```
zephyr/drivers/flash/spi_nor.c:388:47: error: 'DT_JEDEC_SPI_NOR_0_ERASE_BLOCK_SIZE' undeclared here (not in a function)
zephyr/drivers/flash/spi_nor.c:409:22: error: 'DT_JEDEC_SPI_NOR_0_WRITE_BLOCK_SIZE' undeclared here (not in a function);
```

**To Reproduce**
Steps to reproduce the behavior:
1. modify my board dts, add spi-nor flash tree:
```
&spi2 {
	status = "ok";
	sck-pin = <4>;
	mosi-pin = <5>;
	miso-pin = <13>;

	cs-gpios = <&gpio0 15 0>;

	w25x16vsi0: w25x16vsi@0 {
		compatible = "jedec,spi-nor";
		reg = <0>;
		spi-max-frequency = <2000000>;
		label = "W25X16VSI";
		jedec-id = <0xEF 0x80 0x3a>;
		size = <0x1000000>;
		erase-block-size = <0x10000>;
		write-block-size = <1>;
		hold-gpios = <&gpio0 9 0>;
	};
};
```
2. modify my board default conf，add spi-nor flash config
```
CONFIG_SPI_2=y
CONFIG_SPI_2_NRF_SPI=y
CONFIG_FLASH=y
CONFIG_SPI_NOR=y
CONFIG_SPI_NOR_CS_WAIT_DELAY=5
CONFIG_SPI_NOR_PAGE_SIZE=256
CONFIG_SPI_NOR_SECTOR_SIZE=4096
CONFIG_FLASH_HAS_PAGE_LAYOUT=y
CONFIG_FLASH_PAGE_LAYOUT=y
```
3. make, you can find undeclared DT_JEDEC_SPI_NOR_0_ERASE_BLOCK_SIZE & DT_JEDEC_SPI_NOR_0_WRITE_BLOCK_SIZE

**Expected behavior**
Build Pass & Spi-nor work well

**Additional context**
If the spi-flash be put in chosen, the DT_JEDEC_SPI_NOR_0_ERASE_BLOCK_SIZE & DT_JEDEC_SPI_NOR_0_WRITE_BLOCK_SIZE will be create
```
	chosen {
		zephyr,console = &uart0;
		zephyr,shell-uart = &uart0;
		zephyr,sram = &sram0;
		zephyr,flash = &w25x16vsi0;
		//zephyr,flash = &flash0;
	};
```
but, remove the internal flash0 will cause other build error

--- a/drivers/flash/spi_nor.c
+++ b/drivers/flash/spi_nor.c
@@ -355,7 +355,7 @@ static int spi_nor_configure(struct device *dev)
 		return -ENODEV;
 	}
 
-	data->cs_ctrl.gpio_pin = DT_JEDEC_SPI_NOR_0_HOLD_GPIOS_PIN;
+	data->cs_ctrl.gpio_pin = DT_JEDEC_SPI_NOR_0_CS_GPIO_PIN;
 	data->cs_ctrl.delay = CONFIG_SPI_NOR_CS_WAIT_DELAY;
 
 	data->spi_cfg.cs = &data->cs_ctrl;

--- a/dts/bindings/mtd/jedec,spi-nor.yaml
+++ b/dts/bindings/mtd/jedec,spi-nor.yaml
@@ -23,6 +23,20 @@ properties:
     description: JEDEC ID as manufacturer ID, memory type, memory density
     generation: define
 
+  erase-block-size:
+   type: int
+   description: address alignment required by flash erase operations
+   generation: define
+   category: optional
+   label: alignment
+
+  write-block-size:
+   type: int
+   description: address alignment required by flash write operations
+   generation: define
+   category: optional
+   label: alignment
+
   size:
     type: int
     category: optional

